src/api/app/services/trigger_controller_service/scm_extractor.rb

```diff
@@ -99,5 +97,41 @@ def gitlab_api_endpoint(http_url)
       uri = URI.parse(http_url)
       "#{uri.scheme}://#{uri.host}"
     end
+
+    def github_payload_push(payload)
+      payload_ref = @payload.fetch(:ref, '')
+      payload.merge!({
+                       # We need this for Workflow::Step#branch_request_content_github
+                       source_repository_full_name: @payload.dig(:repository, :full_name),
+                       # We need this for SCMStatusReporter#call
+                       target_repository_full_name: @payload.dig(:repository, :full_name),
+                       ref: payload_ref,
+                       # We need this for Workflow::Step#branch_request_content_{github,gitlab}
+                       commit_sha: @payload[:after],
+                       # We need this for Workflows::YAMLDownloader#download_url
+                       # base_ref is nil when the push event is for commits
+                       target_branch: (@payload[:base_ref] || payload_ref).sub('refs/heads/', '')
```

Comment:
@dmarcoux can we rely on the `:base_ref`, I saw its in the payload, but somehow not really mentioned in the github docs.

Reply:
Well, should be fine, its their, it indicates the target